src/modules/storage.ts

```diff
@@ -0,0 +1,100 @@
+/**
+ * Implements Web Storage API if localStorage is not available.
+ * So it can be available in node.js environment
+ */
+class CatcherStorage implements Storage {
+  private items: { [key: string]: string } = {};
+
+  /**
+   * Returns the value of the item with the specified key.
+   *
+   * @param key - The key of the item you want to retrieve
+   */
+  public getItem(key: string): string | null {
+    if (this.isLocalStorageAvailable()) {
+      return localStorage.getItem(key);
+    }
+
+    return this.items[key] || null;
+  }
+
+  /**
+   * Sets the value of the pair identified by key to value, creating a new key/value pair if none existed for key previously.
+   *
+   * @param key - The key of the item you want to store
+   * @param value - The value of the item you want to store
+   */
+  public setItem(key: string, value: string): void {
```

Comment:
Lets make this class more abstract:

```ts
export interface HawkStorageLike {
  getItem: (key: string) => string | undefined;
  setItem(key: string) => boolean;
}

export class CatcherStorage {
  constructor(private readonly storage: HawkStorageLike){}
  
  getItem(){
    return this.storage.getItem()
  }
  
  setItem(){
    return this.storage.setItem()
  }
}
```

and catcher will set available implementation:

```ts
// catcher.ts

prepateStorage(){
  const isLocalStorageAvailable = ...
  const storageImplementation = isLocalStorageAvailable ? LocalStorage : CustomStorage

  this.storage = new CatcherStorage(storageImplementation)
}
```

and user can pass own implementation through the catcher initial settings:

```ts
import { HawkCatcher, type HawkStorageLike } from '@hawk.so/javascript'

class ReactNativeStorage implements HawkStorageLike {
 ...
}

new HawkCatcher {
  //...
  storage: new ReactNativeStorge()
}

```